vortex-io/src/write.rs

```diff
@@ -3,13 +3,34 @@ use std::io::{self, Cursor, Write};
 
 use vortex_buffer::io_buf::IoBuf;
 
+const ZEROS: [u8; 512] = [0_u8; 512];
+
 pub trait VortexWrite {
+    #[allow(async_fn_in_trait)]
+    async fn write_zeros(&mut self, mut len: usize) -> io::Result<()> {
+        while len != 0 {
+            let k = std::cmp::min(len, ZEROS.len());
+            self.write_all(&ZEROS[0..k]).await?;
+            len -= k;
+        }
+        Ok(())
+    }
+
     fn write_all<B: IoBuf>(&mut self, buffer: B) -> impl Future<Output = io::Result<B>>;
     fn flush(&mut self) -> impl Future<Output = io::Result<()>>;
     fn shutdown(&mut self) -> impl Future<Output = io::Result<()>>;
 }
 
 impl VortexWrite for Vec<u8> {
+    fn write_zeros(&mut self, mut len: usize) -> impl Future<Output = io::Result<()>> {
+        while len != 0 {
+            let k = std::cmp::min(len, ZEROS.len());
+            self.extend_from_slice(&ZEROS[0..k]);
+            len -= k;
+        }
+        ready(Ok(()))
+    }
+
```

Comment:
This appears sufficient to avoid the ire of the compiler. Without this specialized impl, it fails to compile. I suppose there's something wrong with Vec<u8>'s `write_all`?